components/motor/gpio/motor_encoder.go

```diff
@@ -442,59 +432,25 @@ func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, revolutions float
 	default:
 	}
 
-	currentPos, err := m.position(ctx, nil)
-	if err != nil {
-		return err
-	}
-
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
 
-	m.state.regulated = true
-	if revolutions == 0 {
-		if m.loop != nil {
-			velVal := math.Abs(rpm * m.ticksPerRotation / 60)
-			// when rev = 0, only velocity is controlled
-			// setPoint is +/- infinity, maxVel is calculated velVal
-			if err := m.updateControlBlock(ctx, math.Inf(int(rpm)), velVal); err != nil {
-				return err
-			}
-		} else {
-			// Moving 0 revolutions is a special value meaning "move forever."
-			oldRpm := m.state.goalRPM
-			m.state.goalRPM = rpm
-			m.state.goalPos = math.Inf(int(rpm))
-			// if we are already moving, let rpmMonitor deal with setPower
-			if math.Abs(oldRpm) > 0.001 && m.state.direction == m.directionMovingInLock() {
-				return nil
-			}
-			// if moving from stop, start at 10% power
-			if err := m.setPower(ctx, m.state.direction*0.1, true); err != nil {
-				return err
-			}
-		}
-		return nil
-	}
-
-	goalPos := (math.Abs(revolutions) * m.ticksPerRotation * m.state.direction) + currentPos
-
-	m.state.goalRPM = math.Abs(rpm) * m.state.direction
-	m.state.goalPos = goalPos
-
-	if m.loop != nil {
+	switch {
+	case m.loop != nil:
 		velVal := math.Abs(rpm * m.ticksPerRotation / 60)
-		// when rev is not 0, velocity and position are controlled
-		// setPoint is goalPos, maxVel is calculated velVal
+		// when rev = 0, only velocity is controlled
+		// setPoint is +/- infinity, maxVel is calculated velVal
 		if err := m.updateControlBlock(ctx, goalPos, velVal); err != nil {
 			return err
 		}
-	} else {
-		startingPwr := 0.1 * m.state.direction
-		err = m.setPower(ctx, startingPwr, true)
-		if err != nil {
+	case m.loop == nil:
+		// start at 10% power
+		if err := m.setPower(ctx, direction*0.1, true); err != nil {
```

Comment:
[note to self] this sets an initial power to the motor based on the direction passed into `goForInternal`, which effectively sets the initial state defined motor direction that gets used by `rpmMonitor`.

Reply:
yep, basically just gives an initial setPower call that is (hopefully) enough to actually start the motor moving since some (especially those on the viam-rover) take a bit of power to move from rest